crates/pink/sidevm/sidevm/src/net.rs

```diff
@@ -0,0 +1,162 @@
+//! Networking support.
+
+use std::future::Future;
+use std::io::Error;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+use hyper::server::accept::Accept;
+use tokio::io::{AsyncBufRead, AsyncRead, AsyncWrite};
+
+use crate::env::{self, Result};
+use crate::{ocall, ResourceId};
+
+/// A TCP socket server, listening for connections.
+pub struct TcpListener {
+    res_id: ResourceId,
+}
+
+/// A connected TCP socket.
+#[derive(Debug)]
+pub struct TcpStream {
+    res_id: ResourceId,
+    buf: [u8; 32],
```

Comment:
It is a bit small. But if the App handles tons of TCP connections concurrently, it would waste memory to increase this buffer.

Consider split the buffer out to a wrapper like `BufferedTcpStrem` with a configurable buffer size.

Reply:
OK. We can go with that first

Reply:
Added a TODO there, and I'll do it in near future.